### PR TITLE
Wrap txpool sender

### PR DESCRIPTION
**Pull request by rakita**
Wrap txpool mpsc::Sender around Sender structure that gives us better ergonomic in calling functionalities from txpool,

related to: https://github.com/FuelLabs/fuel-core/issues/429

### CODE REVIEW

**Review by ControlCplusControlV — APPROVED** [review]
LGTM

**Review by Voxelot — APPROVED** [review]

**Comment by Voxelot** [comment]
thanks for cleaning up the `.await.await`'s 😂

File: fuel-core-interfaces/src/txpool.rs [hunk 1]
```diff
@@ -26,6 +27,65 @@ pub trait TxPoolDb:
     }
 }
 
+#[derive(Clone, Deref, DerefMut)]
+pub struct Sender(mpsc::Sender<TxPoolMpsc>);
```
[the remark this answers is not in the document]
Comment (rakita) [reply fuel-core-interfaces/src/txpool.rs:31]: I like this naming, will make the change.

[the remark this answers is not in the document]
Comment (rakita) [reply fuel-core-interfaces/src/txpool.rs:31]: Why is moving away from channel terminology a good thing when this is just a wrapper and channels are async primitives? As with `Client`, we are adding new abstraction naming when in essence it is `mpsc::Sender` inside.
I made the change here, but while coding the naming felt weird. https://github.com/FuelLabs/fuel-core/commit/288ccef4a7bf4ca6548e07d03fcb933089864291

**Pull request merged**